From 9f85a725c5b2e7c164e9c3d9b0bb0fcfdda41e5a Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 2 Nov 2017 15:39:16 +0900
Subject: [PATCH 374/909] media: i2c: adv748x: Fix get_format function

commit 09a84effe388f8fdea8682ca6a14c84c98a4b1f4 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

When calling the VIDIOC_SUBDEV_S_FMT ioctl, the resolution value
may be not reflected.
This patch adds resolution information query procedure in set_fmt
callback function for corresponding its problem.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv748x/adv748x-afe.c  | 4 ++++
 drivers/media/i2c/adv748x/adv748x-hdmi.c | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/drivers/media/i2c/adv748x/adv748x-afe.c b/drivers/media/i2c/adv748x/adv748x-afe.c
index edd25e895e5d..ed83e12c8221 100644
--- a/drivers/media/i2c/adv748x/adv748x-afe.c
+++ b/drivers/media/i2c/adv748x/adv748x-afe.c
@@ -352,6 +352,7 @@ static int adv748x_afe_get_format(struct v4l2_subdev *sd,
 {
 	struct adv748x_afe *afe = adv748x_sd_to_afe(sd);
 	struct v4l2_mbus_framefmt *mbusformat;
+	v4l2_std_id std = 0;
 
 	/* It makes no sense to get the format of the analog sink pads */
 	if (sdformat->pad != ADV748X_AFE_SOURCE)
@@ -361,6 +362,9 @@ static int adv748x_afe_get_format(struct v4l2_subdev *sd,
 		mbusformat = v4l2_subdev_get_try_format(sd, cfg, sdformat->pad);
 		sdformat->format = *mbusformat;
 	} else {
+		/* Set std_id automatically */
+		adv748x_afe_querystd(sd, &std);
+		adv748x_afe_s_std(sd, std);
 		adv748x_afe_fill_format(afe, &sdformat->format);
 		adv748x_afe_propagate_pixelrate(afe);
 	}
diff --git a/drivers/media/i2c/adv748x/adv748x-hdmi.c b/drivers/media/i2c/adv748x/adv748x-hdmi.c
index aecc2a84dfec..17e38b8e0d14 100644
--- a/drivers/media/i2c/adv748x/adv748x-hdmi.c
+++ b/drivers/media/i2c/adv748x/adv748x-hdmi.c
@@ -438,6 +438,11 @@ static int adv748x_hdmi_get_format(struct v4l2_subdev *sd,
 		mbusformat = v4l2_subdev_get_try_format(sd, cfg, sdformat->pad);
 		sdformat->format = *mbusformat;
 	} else {
+		struct v4l2_dv_timings timings;
+
+		adv748x_hdmi_query_dv_timings(&hdmi->sd, &timings);
+		hdmi->timings = timings;
+
 		adv748x_hdmi_fill_format(hdmi, &sdformat->format);
 		adv748x_hdmi_propagate_pixelrate(hdmi);
 	}
-- 
2.17.1

